src/classes/GE_TemplateBuilderCtrl.cls

```diff
@@ -0,0 +1,155 @@
+    * @description Method that retrieves all Field Mapping Sets in an org. Used in the Gift Fields tab's
+    * Field Mapping Sets combobox in the Form Template Builder. Field Mapping Set selection determines
+    * which Object Groups and Field Mappings are available to the Form Builder.
+    */
+    @AuraEnabled(cacheable=true)
+    public static Data_Import_Field_Mapping_Set__mdt[] getFieldMappingSetNames() {
+        return [SELECT Id, DeveloperName, MasterLabel FROM Data_Import_Field_Mapping_Set__mdt];
+    }
+
+    /*******************************************************************************************************
+    * @description Method collects object mappings and field mappings using BDI_FieldMappingCustomMetadata
+    * based on a field mapping set developer name. Creates ObjectMappingWrappers and CheckboxWrappers out
+    * of the object mapping and field mappings respectively. These wrappers are easier to use in the UI. No
+    * need to worry about namespaces in the fields.
+    */
+    @AuraEnabled
+    public static ObjectMappingWrapper[] getFieldAndObjectMappingsByFieldMappingSetName(String fieldMappingSetName) {
+        bdiCMT = BDI_FieldMappingCustomMetadata.getInstance(fieldMappingSetName);
+        ObjectMappingWrapper[] objectWrappers = new ObjectMappingWrapper[]{};
+
+        for (Data_Import_Object_Mapping__mdt objectMapping : bdiCMT.objMappingsByDevName.values()) {
+            ObjectMappingWrapper omw = new ObjectMappingWrapper(objectMapping);
+
+            if (objectMapping.Data_Import_Field_Mappings__r != null) {
+                for (Data_Import_Field_Mapping__mdt fieldMapping : objectMapping.Data_Import_Field_Mappings__r) {
+                    CheckboxWrapper fieldMappingWrapper = new CheckboxWrapper(
+                        fieldMapping.MasterLabel,
+                        fieldMapping.DeveloperName,
+                        false
+                    );
+                    omw.fieldMappingCheckboxes.add(fieldMappingWrapper);
+                }
+            }
+
+            objectWrappers.add(omw);
+        }
+
+        return objectWrappers;
+    }
+
+    /*******************************************************************************************************
+    * @description Method checks the Data Import Batch object's field describes and returns a list of
+    * CheckboxWrappers for use in the UI's Batch Header tab.
+    */
+    @AuraEnabled(cacheable=true)
+    public static CheckboxWrapper[] getBatchFields() {
+        CheckboxWrapper[] wrappers = new CheckboxWrapper[]{};
+
+        Map<String, Schema.DescribeFieldResult> fields = UTIL_Describe.getAllFieldsDescribe(SObjectType.DataImportBatch__c.Name);
+        for (String key : fields.keySet()) {
+            wrappers.add(new CheckboxWrapper(fields.get(key)));
+        }
+        return wrappers;
+    }
+
+    public class ObjectMappingWrapper {
```

Comment:
Comparing this to BDI_ManageAdvancedMappingCtrl.DataImportObjectMappingWrapper, which has most of these properties (although many additional) - does it make sense to have multiple wrapper classes defined?

Reply:
Yeah, I was mulling this over as well. I ultimately went with just dropping in a smaller wrapper in this controller for ease and because I didn't want to tie into the BDI_ManageAdvancedMappingCtrl. If we were to utilize one wrapper, should we move it to something like a UTIL_MappingWrappers class or something along those lines? Have one location for all the shared object/field mapping wrappers.

Reply:
@ptewson-sfdo @psadie-sfdo  I've been thinking of creating a common set of wrappers for the Data Import Object and Field Mapping CMT objects (which I mentioned in the n number of object design) to make all the code less object dependent, and potentially they could be used here and  BDI_ManageAdvancedMappingCtrl as well as all of the BDI_AdditionalObjectService related code.  The only real issue is that as you mentioned not all of the attributes are relevant or necessary in every use case, which could potentially be misleading, but overall I think it would make sense to share.  What do you guys think?

Reply:
I'd vote to have the shared wrappers even if we end up with unused attributes in different use cases.

Reply:
Given that we have a plan to move forward with this, and that WI is separate and will be delivered future, I'm going to mark this resolved with the assumption that the initial merge uses his inner wrapper class, but that the follow up delivery will do the necessary refactor to unify to a single ObjectWrapper class across the system.